src/Core/Repositories/ExpandedPackageRepository.cs

```diff
@@ -95,7 +98,12 @@ public IEnumerable<IPackage> FindPackagesById(string packageId)
                 if (SemanticVersion.TryParse(versionDirectoryName, out version) &&
                     Exists(packageId, version))
                 {
-                    yield return GetPackageInternal(packageId, version);
+                    var package = GetPackageInternal(packageId, version);
+                    if (package == null)
```

Comment:
Actually move the try/catch over here and get rid of the null as a possible value